src/Range.cpp

```diff
@@ -229,9 +266,26 @@ Range Range::intersect (const Range& other) const
     return result;
   }
 
+  // If there is an intersection but no overlap, we have a zero-width
+  // interval [p, p) and another interval [p, q), where q >= p.
+  if (intersects (other))
+    return Range {start, start};
+
   return Range {};
 }
 
+////////////////////////////////////////////////////////////////////////////////
+bool Range::intersects (const Range &other) const
```

Comment:
Checks whether `A ∩ B ≠ ∅`. Similar to `overlap` but allows for intersections of the form `[p, p) ∩ [p, q)`.